### PR TITLE
fix(smart-wallet): handling payments in case of failure to create invitation

### PR DESCRIPTION
**Pull request by dckc**
refs: #9239

## Description

test handling payments in case of failure to create invitation; incorporate fix from ...

 - #9240

### Security / Scaling / Documentation

n/a

### Upgrade Considerations

deployment involves upgrading walletFactory (see #9238) 

### Testing Considerations

swingsetTest (and thus walletDriver) doesn't provide access to balances, so bootstrap test was removed from this PR

### CODE REVIEW

File: packages/boot/test/bootstrapTests/test-wallet-invitation.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    proposal: { give: { In } },
+  };
+  await t.throwsAsync(minter.executeOffer(offer1), {
+    message: /^target has no method/,
+  });
+
+  // TODO: check the ATOM balance; we stub the go code that
+  // tracks it, so it's not clear now to do so.
+
```
Comment (Chris-Hibbert) [inline packages/boot/test/bootstrapTests/test-wallet-invitation.js]: I don't see how this test is useful until this TODO is resolved. Is the test worth adding?

Comment (dckc) [reply packages/boot/test/bootstrapTests/test-wallet-invitation.js]: good question... let's see if I can do an e2e / a3p test...

Comment (dckc) [reply packages/boot/test/bootstrapTests/test-wallet-invitation.js]: I'm punting it to #9238 for now

**Review by Chris-Hibbert — APPROVED** [review]

**Comment by mhofman** [comment]
Kicking out of the merge queue so #9204 can land.

**Comment by mhofman** [comment]
Should this PR include bits of #9238 to update wallet factory ?

**Comment by dckc** [comment]
> Should this PR include bits of #9238 to update wallet factory ?

any particular reason?

seems arbitrary which one has it. the historical coin toss went the other way

**Pull request merged**